ffi/src/sspi/common.rs

```diff
@@ -25,10 +25,15 @@ use crate::utils::into_raw_ptr;
 pub unsafe extern "system" fn FreeCredentialsHandle(ph_credential: PCredHandle) -> SecurityStatus {
     check_null!(ph_credential);
 
-    let cred_handle = (*ph_credential).dw_lower as *mut CredentialsHandle;
+    // SAFETY: `ph_credentials` is not null. We've checked for this above.
+    let cred_handle = unsafe { (*ph_credential).dw_lower as *mut CredentialsHandle };
     check_null!(cred_handle);
 
-    let _cred_handle = Box::from_raw(cred_handle);
+    // SAFETY: `cred_handle` is not null. We've checked for this above.
+    // We create and allocate credentials handles using `Box::into_raw`. Thus,
+    // it is safe to deallocate them using `Box::from_raw`.
+    // The user have to ensure that the credentials handle was created by us.
+    let _cred_handle = unsafe { Box::from_raw(cred_handle) };
```

Comment:
> Address each safety requirement individually, using a bullet list.

Also, "the user have to ensure that the credentials handle was created by us." is more a `# Safety` section that should be in the documentation of the function, as it express a precondition of the `FreeCredentialsHandle` function. In turn, this precondition should be used to justify that the handle was indeed allocated using `Box`.

> If a precondition must be upheld by the caller, mark the function as unsafe and document it with a # Safety section listing the invariants.